contracts/contribute/ContributeManager.sol

```diff
@@ -448,7 +452,7 @@ contract ContributeManager is ERC721TokenReceiver {
             IStaking(curStakingInstance).unstake(serviceId);
 
             // Approve service NFT for the staking instance
-            INFToken(serviceRegistry).approve(curStakingInstance, serviceId);
```

Comment:
Instead of doing something this ugly, why not just merge CM into the Contributors? Being a stand-alone ownerless contract was the sole purpose of it. If not - why even bother with periphery?